### DISCUSSION TITLE
Add device class for sensor monetary

### DISCUSSION THREAD

**Discussion by frenck** [home-assistant/architecture#539] [Entity Models] [answered]
## Context

Some integrations display monetary values, it would be nice to display those better.

## Proposal

Add a device class `DEVICE_CLASS_MONETARY`, for measuring an amount of money/currency.

We currently already have the following units available:

```py
CURRENCY_EURO = "€"
CURRENCY_DOLLAR = "$"
CURRENCY_CENT = "¢"
```

For the frontend, it would use icon: `mdi:cash`

![image](https://user-images.githubusercontent.com/195327/112864546-30ada280-90b8-11eb-8e5c-7e19ae9b41d9.png)

If possible, we could use the MDI icons that matches the given unit of measurement (e.g., using the dollar or euro icons).

This can be used by virtually any financial categorized integration: https://www.home-assistant.io/integrations/#finance
Additionally there are other integrations that can use this, like Toon, which can display daily power/gas usage in a monetary value.

## Consequences

Needs to be implemented.

**Comment by emontnemery** [comment]
`DEVICE_CLASS_MONETARY` is too broad, `DEVICE_CLASS_MONETARY_AMOUNT`, `DEVICE_CLASS_MONETARY_VALUE` or `DEVICE_CLASS_MONETARY_SUM` would be more precise to make it clear it's an amount of money which is being measured.

**Reply by frenck** [reply]
Aren't the amount and value part of the entity state?
I'm not sure why sum matters in this case? A sum can still be current?

**Reply by emontnemery** [reply]
I meant to pick one of the three. Just 'monetary' could be something else, a unit price for example.

**Reply by frenck** [reply]
A unit price could be current as well?
We use device class at this point, to represent the type of data it represents, e.g., temperature, humidity, or a monetary value. The main goal is to adjust the UI for it. What type of monetary value or temperature, humidity is stored (sum, average, current, history, yesterday, tomorrow, forcasting) has never been a responsibility of a device class.

**Reply by emontnemery** [reply]
Sure, but an amount of money ($, €, £) is not same as a unit rate ($/h, €/kWh, £/m³), it's like distance vs velocity, but both are monetary related.

**Reply by thomasloven** [reply]
Are those common enough to warrant reclassifying the base case?
Wouldn't it be better then to have `DEVICE_CLASS_MONETARY` as a broad class by design and then append further classifiers if found necessary, such as `DEVICE_CLASS_MONETARY_RATE`?

**Reply by frenck** [reply]
How is that different from e.g., temperature? :)
Or the velocity of the temperature rising in my bathroom?

(Honestly playing a devils card now, but it is a generic problem imho).

**Reply by emontnemery** [reply]
Well, "temperature" is the temperature in your bathroom, measured in °C or °F, the velocity of the temperature rising is not a temperature, it's a temperature rate measured in "°C/s".

(Continuing on the silly path)

**Reply by emontnemery** [reply]
@thomasloven for the energy tracking we will want to keep track of an amount of money (accumulated electricity cost in € for example) and the rate (€/kWh).

But yeah, the most common case must be to keep track of an amount of money (savings account, dogecoin mined, electricity cost, etc.), and the electricity unit rate or dogecoins mined/h (€/kWh, Ɖ/h) will probably be less common.

**Comment by emontnemery** [comment, marked as answer]
Backend support implemented in https://github.com/home-assistant/core/pull/52087

**Comment by Mariusthvdb** [comment]
please consider adjusting the icon to local settings, (and not default the cash icon, which seems very US centered) to give you an idea in Euro's,  I posted this in discord beta:

![](https://cdn.discordapp.com/attachments/427516175237382144/870278765001990154/Schermafbeelding_2021-07-29_om_14.16.55.png)

btw the above is using the `device_class: monetary` with `unit_of_measurement: EUR`, (note the unit preceding the state)

while  `unit_of_measurement: €` makes the unit follow the state:

<img width="420" alt="Schermafbeelding 2021-07-29 om 15 25 40" src="https://user-images.githubusercontent.com/33354141/127500750-dc3eafa6-358f-489f-af6f-bb7ad6de2bba.png">

**Reply by thomasloven** [reply]
Localized icons are probably not the way to go. Lots of currencies don't even have any.

**Reply by Mariusthvdb** [reply]
suppose you're right. in that case we probably shouldn't set the dollar biljet to be the default cash icon for everyone either. cash-register would be more appropriate ;-) 

amazed there isn't a generic coin icon

**Comment by Mariusthvdb** [comment]
any thoughts on the position of the unit ? seems so unique compared to all other units, to precede the state (and too close too)
